readthedocs/projects/tasks.py

```diff
@@ -412,36 +412,6 @@ def run(
             # Send notifications for unhandled errors
             self.send_notifications(version_pk, build_pk)
             return False
-        else:
-            # No exceptions in the setup step, catch unhandled errors in the
-            # build steps
-            try:
-                self.run_build(docker=docker, record=record)
-            except Exception:
-                log.exception(
-                    'An unhandled exception was raised during project build',
-                    extra={
-                        'stack': True,
-                        'tags': {
-                            'build': build_pk,
-                            'project': self.project.slug,
-                            'version': self.version.slug,
-                        },
-                    },
-                )
-                if self.build_env is not None:
-                    self.build_env.failure = BuildEnvironmentError(
```

Comment:
If the failure happens in the `self.build_env` we are loosing this exception and not marking the build as finished with error.

We could probably make this check in the above `except` block together with `self.setup_env`.

Reply:
Yep, missed this one